CreateRedirectConfigurationServlet throws Exception when ancestor node types are different than expected

## Required Information

* [x] AEM Version, including Service Packs, Cumulative Fix Packs, etc: 2024.7.17098.20240711T134106Z
* [X] ACS AEM Commons Version: 6.6.2
* [X] Reproducible on Latest? yes


### Expected Behavior

Creating redirects should work also when the ancestor CAC nodes have an unexpected node type

### Actual Behavior

The following exception can be seen:
```
4.07.2024 08:36:54.158 [cm-p124459-e1227496-aem-author-7656cddfd4-pnh4w] *ERROR* [x.x.x.x [1721810214111] POST /apps/acs-commons/content/redirect-manager/redirects/_jcr_content.create.html HTTP/1.1] com.adobe.acs.commons.redirects.servlets.CreateRedirectConfigurationServlet failed to create configuration org.apache.sling.api.resource.PersistenceException: Unable to commit changes to session. at org.apache.sling.jcr.resource.internal.helper.jcr.JcrResourceProvider.commit(JcrResourceProvider.java:619) [org.apache.sling.jcr.resource:3.3.2] at org.apache.sling.resourceresolver.impl.providers.stateful.AuthenticatedResourceProvider.commit(AuthenticatedResourceProvider.java:247) [org.apache.sling.resourceresolver:1.11.6] at org.apache.sling.resourceresolver.impl.helper.ResourceResolverControl.commit(ResourceResolverControl.java:531) [org.apache.sling.resourceresolver:1.11.6] at org.apache.sling.resourceresolver.impl.ResourceResolverImpl.commit(ResourceResolverImpl.java:1030) [org.apache.sling.resourceresolver:1.11.6]
...
Caused by: org.apache.jackrabbit.oak.api.CommitFailedException: OakConstraint0001: /conf/hcp-plus/settings[[nt:folder]]: No matching definition found for child node redirects with effective type [nt:unstructured] at org.apache.jackrabbit.oak.plugins.nodetype.TypeEditor$1.onConstraintViolation(TypeEditor.java:109) [org.apache.jackrabbit.oak-core:1.66.0]
```

### Steps to Reproduce

Just have a CAC entry node `settings` of type `nt:folder` and then try to create redirects below.

### Proposed Fix

Create the redirects node as type `sling:Folder` instead of `nt:unstructured` because the former can be created below `nt:folder` while the latter cannot. The type is set in https://github.com/Adobe-Consulting-Services/acs-aem-commons/blob/8c3533026031876a0437f526cbc817e51e581928/bundle/src/main/java/com/adobe/acs/commons/redirects/servlets/CreateRedirectConfigurationServlet.java#L100

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -14,6 +14,10 @@ The format is based on [Keep a Changelog](http://keepachangelog.com)
 
 - #3390 - Remove usage of commons collections 3
 
+### Fixed
+
+- #3398 - CreateRedirectConfigurationServlet throws PersistenceException when ancestor node types are different than expected
+
 ## 6.6.2 - 2024-06-25
 
 ### Fixed

--- a/bundle/src/main/java/com/adobe/acs/commons/redirects/servlets/CreateRedirectConfigurationServlet.java
+++ b/bundle/src/main/java/com/adobe/acs/commons/redirects/servlets/CreateRedirectConfigurationServlet.java
@@ -30,6 +30,7 @@ import org.apache.sling.api.resource.Resource;
 import org.apache.sling.api.resource.ResourceResolver;
 import org.apache.sling.api.resource.ResourceUtil;
 import org.apache.sling.api.servlets.SlingAllMethodsServlet;
+import org.apache.sling.jcr.resource.api.JcrResourceConstants;
 import org.osgi.service.component.annotations.Component;
 import org.osgi.service.component.annotations.Reference;
 import org.osgi.service.component.annotations.ReferenceCardinality;
@@ -89,7 +90,7 @@ public class CreateRedirectConfigurationServlet extends SlingAllMethodsServlet {
             Resource bucket = root.getChild(bucketName);
             if (bucket == null) {
                 bucket = resolver.create(root, bucketName,
-                        ImmutableMap.of(JcrConstants.JCR_PRIMARYTYPE, "sling:Folder"));
+                        ImmutableMap.of(JcrConstants.JCR_PRIMARYTYPE, JcrResourceConstants.NT_SLING_FOLDER));
                 log.info("created {}", bucket.getPath());
             }
 
@@ -97,7 +98,7 @@ public class CreateRedirectConfigurationServlet extends SlingAllMethodsServlet {
             if (config == null) {
                 String contextPrefix = StringUtils.defaultString(request.getParameter(REQ_PARAM_CTX_PREFIX));
                 config = resolver.create(bucket, configName,
-                        ImmutableMap.of(JcrConstants.JCR_PRIMARYTYPE, JcrConstants.NT_UNSTRUCTURED,
+                        ImmutableMap.of(JcrConstants.JCR_PRIMARYTYPE, JcrResourceConstants.NT_SLING_FOLDER,
                                 ResourceResolver.PROPERTY_RESOURCE_TYPE, REDIRECTS_RESOURCE_PATH,
                                 REQ_PARAM_CTX_PREFIX, contextPrefix));
                 log.info("created {} with context prefix '{}'", config.getPath(), contextPrefix);

